lib/b-tree.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+
+function BTree(degree = DEFAULT_DEGREE) {
+  this.root = new BTreeNode();
+  this.minDegree = degree;
+}
+
+function BTreeNode(leaf = true) {
+  this.elements = [];
+  this.leaf = leaf;
+  this.lastChild = null;
```

Comment:
Why is `lastChild` stored separately?

Reply:
In each element of the node store a reference to a node with elements less than it. Therefore, somewhere in the node, we still need to store a reference to a node with elements larger than the last element in the node. I decided to store it in a separate field.